source/utils/timed-out.ts

```diff
@@ -32,12 +32,23 @@ export default (request: ClientRequest, delays: Delays, options: TimedOutOptions
 	const cancelers: Array<typeof noop> = [];
 	const {once, unhandleAll} = unhandler();
 
+	const cancelTimeouts = (): void => {
+		for (const cancel of cancelers) {
+			cancel();
+		}
+
+		cancelers.length = 0;
+
+		unhandleAll();
+	};
+
 	const addTimeout = (delay: number, callback: (...args: unknown[]) => void, ...args: unknown[]): (typeof noop) => {
 		// Event loop order is timers, poll, immediates.
 		// The timed event may emit during the current tick poll phase, so
 		// defer calling the handler until the poll phase completes.
 		let immediate: NodeJS.Immediate;
 		const timeout: NodeJS.Timeout = setTimeout(() => {
+			cancelTimeouts();
```

Comment:
I don't think it has any sense. It's useless. Once the error is thrown all timeouts are canceled.

Reply:
Unless the error isn't emitted, of course. Or if another listener for the error event did something weird.

Reply:
If it isn't emitted, only one of these two cases could happen:
1. The error was thrown and cancelled all the timeouts.
2. The request was aborted, so no further errors are thrown:

https://github.com/sindresorhus/got/blob/985a3b2d384072bd7ddc9f2f89474cd3a41592a7/source/request-as-event-emitter.ts#L196

I have double checked, it's not possible for the error event listener to behave weird.